### DISCUSSION TITLE
segmented loading

### DISCUSSION THREAD

**Discussion by Sc-Dz** [dlemstra/Magick.NET#1190] [Help]
I have a file 800M, when loaded it takes up about 3G of memory, is it possible to load it in segments and then process it.

![1](https://user-images.githubusercontent.com/106168467/170063010-26f8e3d5-92c0-49e9-a725-60e92118e66f.png)
![2](https://user-images.githubusercontent.com/106168467/170063073-6caff2b9-2066-4ce2-ac49-41ee32b47732.png)

**Comment by dlemstra** [comment]
If you want to read individual frame you could use `yourfile.tif[0]` / `yourfile.tif[1]` or set the `FrameIndex` in the `MagickReadSettings`.

**Reply by Sc-Dz** [reply]
There is only one file。

**Comment by Sc-Dz** [comment]
@dlemstra Have you considered opening Alipay or WeChat donation channels?

**Comment by dlemstra** [comment]
Right now it is not possible to do what you want. You can read part of the image with `ExtractArea` of the `MagickReadSettings` but you will then still read the whole image first.
